pkg/steps/multi_stage.go

```diff
@@ -553,7 +556,10 @@ func (s *multiStageTestStep) runPod(ctx context.Context, pod *coreapi.Pod, notif
 	if _, err := createOrRestartPod(s.podClient.Pods(s.jobSpec.Namespace()), pod); err != nil {
 		return fmt.Errorf("failed to create or restart %q pod: %w", pod.Name, err)
 	}
-	_, err := waitForPodCompletion(ctx, s.podClient.Pods(s.jobSpec.Namespace()), s.eventClient.Events(s.jobSpec.Namespace()), pod.Name, notifier, false)
+	newPod, err := waitForPodCompletion(ctx, s.podClient.Pods(s.jobSpec.Namespace()), s.eventClient.Events(s.jobSpec.Namespace()), pod.Name, notifier, false)
+	if newPod != nil {
+		pod = newPod
```

Comment:
This change (I think) is breaking [lint][1]:

```
 pkg/steps/multi_stage.go:566:139: SA5011: possible nil pointer dereference (staticcheck)
		linksText.WriteString(fmt.Sprintf("Link to step on registry info site: https://steps.ci.openshift.org/reference/%s", strings.TrimPrefix(pod.Name, s.name+"-")))
```

I dunno why it is not noticing me only clobbering `pod` when `newPod` is non-`nil`.  But I guess I could add a `pod != nil` guard closer to those `WriteString` calls to make it more obvious to the linter?

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ci-tools/1257/pull-ci-openshift-ci-tools-master-lint/1314290804134514688#1:build-log.txt%3A13

Reply:
The reason is slightly different. The linter (very likely, this is a common static analysis technique) infers the fact that `pod` may be `nil` on L566 from the fact that you added a `pod != nil` check **later**, on L572. If the check makes sense on L572, it means that `pod` can already be `nil` earlier.